src/allocator.rs

```diff
@@ -4,8 +4,56 @@ use crate::reduction::EvalErr;
 use chia_bls::{G1Element, G2Element};
 use clvm_traits::{ClvmDecoder, ClvmEncoder, FromClvmError, ToClvmError};
 
+const MAX_NUM_ATOMS: usize = 62500000;
+const MAX_NUM_PAIRS: usize = 62500000;
+const NODE_PTR_IDX_BITS: u32 = 26;
+const NODE_PTR_IDX_MASK: u32 = (1 << NODE_PTR_IDX_BITS) - 1;
+
 #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct NodePtr(pub i32);
+pub struct NodePtr(u32);
+
+enum ObjectType {
+    Pair,
+    Bytes,
+}
+
+// The top 6 bits of the NodePtr indicate what type of object it is
+impl NodePtr {
+    pub fn null() -> Self {
```

Comment:
I asked ChatGPT and it suggests that in lisp `nil` is used more commonly than `null`.

Reply:
yeah, I think that has been established. we still use `null()` though, and I think changing that (everywhere) ought to be a separate PR

Reply:
https://github.com/Chia-Network/clvm_rs/pull/360